azurerm/helpers/resourceproviders/registration.go

```diff
@@ -0,0 +1,58 @@
+package resourceproviders
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2017-05-10/resources"
+)
+
+func DetermineResourceProvidersRequiringRegistration(availableResourceProviders []resources.Provider, requiredResourceProviders map[string]struct{}) map[string]struct{} {
```

Comment:
Nit: Exported functions should be documented

Reply:
done 👍